internal/tofu/context_plan2_test.go

```diff
@@ -4286,6 +4286,407 @@ import {
+}
+
+import {
+  to   = test_object.a[data.test_object.reference.test_string]
+  id   = "123"
+}
+`,
+			},
+		},
+	}
+
+	for _, configuration := range configurations {
+		t.Run(configuration.Description, func(t *testing.T) {
+			addr := mustResourceInstanceAddr(configuration.ResolvedAddress)
+			m := testModuleInline(t, configuration.inlineConfiguration)
+
+			p := &MockProvider{
+				GetProviderSchemaResponse: &providers.GetProviderSchemaResponse{
+					Provider: providers.Schema{Block: simpleTestSchema()},
+					ResourceTypes: map[string]providers.Schema{
+						"test_object": providers.Schema{Block: simpleTestSchema()},
+					},
+					DataSources: map[string]providers.Schema{
+						"test_object": providers.Schema{
+							Block: &configschema.Block{
+								Attributes: map[string]*configschema.Attribute{
+									"test_string": {
+										Type:     cty.String,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			hook := new(MockHook)
+			ctx := testContext2(t, &ContextOpts{
+				Hooks: []Hook{hook},
+				Providers: map[addrs.Provider]providers.Factory{
+					addrs.NewDefaultProvider("test"): testProviderFuncFixed(p),
+				},
+			})
+			p.ReadResourceResponse = &providers.ReadResourceResponse{
+				NewState: cty.ObjectVal(map[string]cty.Value{
+					"test_string": cty.StringVal("foo"),
+				}),
+			}
+
+			p.ReadDataSourceResponse = &providers.ReadDataSourceResponse{
+				State: cty.ObjectVal(map[string]cty.Value{
+					"test_string": cty.StringVal("bip"),
+				}),
+			}
+
+			p.ImportResourceStateResponse = &providers.ImportResourceStateResponse{
+				ImportedResources: []providers.ImportedResource{
+					{
+						TypeName: "test_object",
+						State: cty.ObjectVal(map[string]cty.Value{
+							"test_string": cty.StringVal("foo"),
+						}),
+					},
+				},
+			}
+
+			plan, diags := ctx.Plan(m, states.NewState(), SimplePlanOpts(plans.NormalMode, testInputValuesUnset(m.Module.Variables)))
+			if diags.HasErrors() {
+				t.Fatalf("unexpected errors\n%s", diags.Err().Error())
+			}
+
+			t.Run(addr.String(), func(t *testing.T) {
+				instPlan := plan.Changes.ResourceInstance(addr)
+				if instPlan == nil {
+					t.Fatalf("no plan for %s at all", addr)
+				}
+
+				if got, want := instPlan.Addr, addr; !got.Equal(want) {
+					t.Errorf("wrong current address\ngot:  %s\nwant: %s", got, want)
+				}
+				if got, want := instPlan.PrevRunAddr, addr; !got.Equal(want) {
+					t.Errorf("wrong previous run address\ngot:  %s\nwant: %s", got, want)
+				}
+				if got, want := instPlan.Action, plans.NoOp; got != want {
+					t.Errorf("wrong planned action\ngot:  %s\nwant: %s", got, want)
+				}
+				if got, want := instPlan.ActionReason, plans.ResourceInstanceChangeNoReason; got != want {
+					t.Errorf("wrong action reason\ngot:  %s\nwant: %s", got, want)
+				}
+				if instPlan.Importing.ID != "123" {
+					t.Errorf("expected import change from \"123\", got non-import change")
+				}
```

Comment:
tiny nitpick: I'm aware that all the test cases right now use an id of `123`, but if they ever changed it could be annoying to have to edit this code too. 

How do you feel about defining the expected id of `123` in the testcases?

Reply:
https://github.com/opentofu/opentofu/pull/1270/commits/c2b98f67597a15b23401363faa2f2c9440cbfecb